test/integration/features/generic-driver.feature

```diff
@@ -0,0 +1,49 @@
+@generic-driver
+Feature: With generic driver Minishift can provision remote unprovisioned VM 
+
+  Scenario: User creates an unprovisioned VM
+    Given setting up environment variable "MINISHIFT_ENABLE_EXPERIMENTAL" with value "y" succeeds
+     When executing "minishift profile set remote" succeeds
+      And executing "minishift start --no-provision" succeeds
+     Then Minishift should have state "Running"
+
+  Scenario: Provisioning remote instance
+    Given unset environment variable "MINISHIFT_ENABLE_EXPERIMENTAL" succeeds
+      And setting scenario variable "REMOTE_VM_IP" to the stdout from executing "minishift ip"
+      And executing "minishift profile set minishift" succeeds
+     When executing "minishift start --vm-driver generic --remote-ipaddress $(REMOTE_VM_IP) --remote-ssh-key .minishift/profiles/remote/machines/remote/id_rsa --remote-ssh-user docker" succeeds
+     Then Minishift should have state "Running"
+
+  Scenario: Stopping with generic driver
+  Stopping with generic driver puts oc cluster down, remote VM however keeps running.
+     When executing "minishift stop" succeeds
+      And executing "minishift status" succeeds
+     Then stdout should match "OpenShift:\s+Stopped"
+
+  Scenario: VM at remote location is still running
+     When executing "minishift status --profile remote" succeeds
+     Then stdout should match "Minishift:\s+Running"
+      And stdout should match "OpenShift:\s+Stopped"
+
+  Scenario: Starting stopped cluster with generic driver
+     When executing "minishift start" succeeds
+     Then executing "minishift status" succeeds
+      And stdout should match "Minishift:\s+Running"
+      And stdout should match "OpenShift:\s+Running"
+  
+  Scenario: Remote VM is running and cluster is up
+     When executing "minishift status --profile remote" succeeds
+     Then stdout should match "Minishift:\s+Running"
+      And stdout should match "OpenShift:\s+Running"
+
+  Scenario: Deleting remote instance
+     When executing "minishift delete --force" succeeds
+     Then Minishift should have state "Does Not Exist"
+
+  Scenario: VM at remote location is still running
+     When executing "minishift profile set remote" succeeds
+     Then Minishift should have state "Running"
+
+  Scenario: Deleting remote VM
```

Comment:
What is the difference between this and `Deleting remote instance` ?

Reply:
@budhram deleting remote instance deletes it via generic driver, which only deletes information about the remote VM from the profile, however the actual VM is still running, generic driver has no control over it. Here in `Deleting remote VM` scenario we changed to profile `remote` so we operate directly on the Minishift VM, this is normal delete via kvm/xhyve/vbox drivers. These few scenarios are here to also check that there are not any interferences between generic driver and VM created with --no-provision flag.

Reply:
we should verify the configuration on the reote host is gone

Reply:
@gbraad good point, can you please guide/give me to some information about the config? Did quick search in repo today but without results.